common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ShuffleIndexInformation.java

```diff
@@ -46,8 +45,11 @@ public ShuffleIndexInformation(File indexFile) throws IOException {
    * Size of the index file
    * @return size
    */
-  public int getSize() {
-    return size;
+  public int getRetainedMemorySize() {
+    // SPARK-33206: here the offsets' capacity is multiplied by 8 as offsets stores long values.
+    // And the extra 176 bytes is the estimate of the `ShuffleIndexInformation` memory footprint
+    // which is relevant in case of small index files (i.e. storing only 2 offsets = 16 bytes).
+    return (offsets.capacity() << 3) + 176;
```

Comment:
Just want to double confirm the number of hard-coded extra bytes since I saw the original number was 1176.

Reply:
960 bytes was the `java.io.File` but now we have an interned `String` (which would not count into the cache):

![image](https://user-images.githubusercontent.com/2017933/155323240-3974f3c1-a2f7-4030-9f48-5e7a7523ec34.png)